firebase-config/src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfig.java

```diff
@@ -358,6 +368,7 @@ public boolean activateFetched() {
    */
   @NonNull
   public String getString(@NonNull String key) {
+    pullArm(key);
```

Comment:
We are essentially doing a write-on-read. Developers are used to treating the get*() APIs as idempotent functions. This will no longer be true. Are you ok if GA counts these as independent events.

Reply:
It seems like we are going to build a variety of products built on top of config that would require this feedback loop to be established. I feel like a filter pattern or a listener pattern is a good candidate to consider here, at the risk of optimizing too early.

Reply:
Yeah, we're OK with GA counting these as separate events. Implemented the listener pattern.